tests/service/test_service.yml

```diff
@@ -0,0 +1,544 @@
+# This test uses skip_host_check, so it will fail if not using
+# FreeIPA version 4.7.0 or later.
+#
+# To test against earlier versions, use test_without_skip_host_check.yml.
+#
+---
+- name: Test service
+  hosts: ipaserver
+  become: yes
+
+  tasks:
+  # setup
+  - name: Get Domain from server name
+    set_fact:
+      ipaserver_domain: "{{ groups.ipaserver[0].split('.')[1:] | join ('.') }}"
+    when: ipaserver_domain is not defined
+
+  - name: Set host1, host2 and svc hosts fqdn
+    set_fact:
+      host1_fqdn: "{{ 'host1.' + ipaserver_domain }}"
+      host2_fqdn: "{{ 'host2.' + ipaserver_domain }}"
+      svc_fqdn: "{{ 'svc.' + ipaserver_domain }}"
+
+  - name: Host absent
+    ipahost:
+      ipaadmin_password: SomeADMINpassword
+      name:
+      - www.ansible.com
+      - no.idontexist.info
+      - svc.ihavenodns.info
+      - "{{ host1_fqdn }}"
+      - "{{ host2_fqdn }}"
+      - "{{ svc_fqdn }}"
+      update_dns: no
+      state: absent
+
+  # If ansible-freeipa dnsrecord module is availble, it can be used.
+  - name: DNS record is not present.
+    command: ipa dnsrecord-del --del-all "{{ ipaserver_domain }}" svc
+    ignore_errors: yes
```

Comment:
Please do not simply use `ignore_errors` like this here as this will still mark the task as failed. Something like

    ignore_errors: yes
    register: result
    failed_when: no
    changed_when: result is defined and
                  "DNS resource record not found" not in result.stderr
 
would be a lot better.

Reply:
Better:

    register: result
    failed_when: result is not defined
    changed_when: result is defined and
                  "DNS resource record not found" not in result.stderr

Reply:
Fixed by removing usage of `ipa dnsrecord-*` and using ansible-freeipa host module.